File: MacAppDemos/ContainerView/ContainerView/MainViewController.m
//
//  MainViewController.m
//  ContainerView
//
//  Created by 李建忠 on 2019/9/22.
//  Copyright © 2019 李建忠. All rights reserved.
//

#import "MainViewController.h"

@interface MainViewController (){
    int i;
}
@property (weak) IBOutlet NSTextField *contentLabel;

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(buttonClicked:) name:@"SidebarVCButtonClickNotification" object:nil];
}

- (void)buttonClicked:(NSNotification *)notification {
    self.contentLabel.stringValue = [NSString stringWithFormat:@"Click %d times",++i];
}

@end

File: MacAppDemos/ContainerViewInSwift/ContainerViewInSwift/MainViewController.swift
//
//  MainViewController.swift
//  ContainerViewInSwift
//
//  Created by 李建忠 on 2019/9/22.
//  Copyright © 2019 李建忠. All rights reserved.
//

import Cocoa

class MainViewController: NSViewController {

    @IBOutlet weak var content: NSTextField!
    private var numberOfClicks = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do view setup here.
        NotificationCenter.default.addObserver(self, selector:#selector(MainViewController.display(notification:)) , name: NSNotification.Name.init(CVNotifications.Click.rawValue), object: nil)
    }
    
    @objc func display(notification: NSNotification) {
        if let contentValue = notification.userInfo?[CVNotificationsUserInfo.Content] as? String , contentValue != "" {
            content.stringValue = contentValue
        } else {
            numberOfClicks += 1
            content.stringValue = "Click \(numberOfClicks) times"
        }
    }
}
